drift-transport-netty/src/main/java/com/facebook/drift/transport/netty/client/ConnectionPool.java

```diff
@@ -69,7 +65,7 @@ public ConnectionPool(
                 .<ConnectionKey, Future<Channel>>removalListener(notification -> closeConnection(notification.getValue()))
                 .build();
 
-        maintenanceThread.scheduleWithFixedDelay(cachedConnections::cleanUp, 1, 1, TimeUnit.SECONDS);
+        scheduledExecutorService.scheduleWithFixedDelay(cachedConnections::cleanUp, 1, 1, TimeUnit.SECONDS);
```

Comment:
In the next commit I'm creating a separate connection pool for each Drift client. To avoid spawning a new thread for each I'm refactoring the ConnectionPool to use a provided scheduled executor